.github/workflows/ci.yml

```diff
@@ -0,0 +1,80 @@
+---
+name: OpenWISP Network Topology CI Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    name: Python==${{ matrix.python-version }} | ${{ matrix.django-version }}
+    runs-on: ubuntu-20.04
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - 3.6
+          - 3.7
+        django-version:
+          - django~=2.2
+          - django~=3.0.0
+          - django~=3.1.0
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install python dependencies
+        run: |
+          pip install -U "pip==20.2.4" wheel setuptools
+          pip install ${{ matrix.django-version }}
+          pip install -U -r requirements-test.txt
+
+      - name: Install openwisp-network-topology
+        run: pip install -e .
+
+      - name: Install test dependencies
+        run: |
+          # needed to test integrations
+          pip install -U https://github.com/openwisp/openwisp-monitoring/tarball/master
+          pip install -U https://github.com/openwisp/openwisp-utils/tarball/master
```

Comment:
Is this really required? Wouldn't it be installed with the rest of the requirements (`requirements.txt`)? :smile:

Reply:
The rest looks good to me

Reply:
hello, @Purhan I tried adding it to `requirements`, but the problem is when openwisp-monitoring is installed it degrades openwisp-utils to 0.7.4(openwisp-monitoring install depencies of openwisp-controller which again install openwisp-utils 0.7.4), Once new release of openwisp-utils is out, this will get resolved I guess.

Reply:
@ManishShah120 Check the build logs. `openwisp-utils` is not being downgraded, it's being upgraded by this line. I ran tests locally and they worked with `0.7.4` but if its necessary to upgrade it, I'd do that in `requirements-test.txt` rather than the build itself. What do you think?

Reply:
Yes, @Purhan I resolved it. Is it OK now??